# This patch modifies the meson.build files to make them work
# with the current MESON syntax.

diff --git a/data/appdata/meson.build b/data/appdata/meson.build
index 1dc4ed7..65a044f 100644
--- a/data/appdata/meson.build
+++ b/data/appdata/meson.build
@@ -1,6 +1,6 @@
 appdata_conf = configuration_data()
 appdata_conf.set('application_id', application_id)
-i18n.merge_file('appdata',
+appdata_file = i18n.merge_file(
   input: configure_file(
     input: 'org.gnome.Dictionary.appdata.xml.in.in',
     output: 'org.gnome.Dictionary.appdata.xml.in',
@@ -18,6 +18,7 @@ if appstream_util.found()
     appstream_util,
     args: [ 'validate', '--nonet', application_id + '.appdata.xml' ],
     workdir: meson.current_build_dir(),
+    depends: appdata_file,
   )
 endif

diff --git a/data/meson.build b/data/meson.build
index 660e6b8..4db85ee 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -3,7 +3,7 @@ subdir('appdata')
 desktop_conf = configuration_data()
 desktop_conf.set('icon', application_id)
 desktop_conf.set('application_id', application_id)
-i18n.merge_file('desktop',
+desktop_file = i18n.merge_file(
   input: configure_file(
     input: 'org.gnome.Dictionary.desktop.in.in',
     output: 'org.gnome.Dictionary.desktop.in',
@@ -22,6 +22,7 @@ if desktop_file_validate.found()
     desktop_file_validate,
     args: application_id + '.desktop',
     workdir: meson.current_build_dir(),
+    depends: desktop_file,
   )
 endif
 
@@ -45,7 +46,7 @@ sources = [
 ]
 
 foreach s: sources
-  i18n.merge_file('sources',
+  i18n.merge_file(
     input: '@0@.in'.format(s),
     output: s,
     install: true,

